tests/plaq_test.cpp

```diff
@@ -0,0 +1,148 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <time.h>
+#include <math.h>
+#include <string.h>
+
+#include <util_quda.h>
+#include <test_util.h>
+#include <dslash_util.h>
+#include "misc.h"
+
+#include <qio_field.h>
+
+#if defined(QMP_COMMS)
+#include <qmp.h>
+#elif defined(MPI_COMMS)
+#include <mpi.h>
+#endif
+
+// In a typical application, quda.h is the only QUDA header required.
+#include <quda.h>
+
+extern bool tune;
+extern int device;
+extern int xdim;
+extern int ydim;
+extern int zdim;
+extern int tdim;
+extern int gridsize_from_cmdline[];
+extern QudaReconstructType link_recon;
+extern QudaReconstructType link_recon_sloppy;
+extern QudaPrecision prec;
+extern QudaPrecision prec_sloppy;
+extern double anisotropy;
+extern double epsilon;
+
+extern bool verify_results;
+
+extern char latfile[];
+extern double gaussian_beta;
+
+extern QudaVerbosity verbosity;
+
+#define MAX(a, b) ((a) > (b) ? (a) : (b))
+
+QudaPrecision cpu_prec = QUDA_DOUBLE_PRECISION;
+QudaPrecision &cuda_prec = prec;
+QudaPrecision &cuda_prec_sloppy = prec_sloppy;
+
+void setGaugeParam(QudaGaugeParam &gauge_param)
+{
+  gauge_param.X[0] = xdim;
+  gauge_param.X[1] = ydim;
+  gauge_param.X[2] = zdim;
+  gauge_param.X[3] = tdim;
+
+  gauge_param.anisotropy = anisotropy;
+  gauge_param.type = QUDA_WILSON_LINKS;
+  gauge_param.gauge_order = QUDA_QDP_GAUGE_ORDER;
+  gauge_param.t_boundary = QUDA_PERIODIC_T;
+
+  gauge_param.cpu_prec = cpu_prec;
+
+  gauge_param.cuda_prec = cuda_prec;
+  gauge_param.reconstruct = link_recon;
+
+  gauge_param.cuda_prec_sloppy = cuda_prec_sloppy;
+  gauge_param.reconstruct_sloppy = link_recon_sloppy;
+
+  gauge_param.gauge_fix = QUDA_GAUGE_FIXED_NO;
+
+#ifdef MULTI_GPU
```

Comment:
While I remember---were we going to move this into the creation of the gauge field itself so we can remove this boilerplate? Is there an issue for this/did it happen already, I missed it, but it can be removed here?

Reply:
There are issues that need to be resolved before the boiler plate can be removed.  Postponed until post 1.0.